### PR TITLE
Delete IndexUtil

### LINKED ISSUE #1254: Delete IndexUtil
`IndexUtil` is defunct. The `IndexReaderUtils` class contains the utility methods that are actively being used (in pyserini) and is well tested.

I'm not aware of active research threads that are using `IndexUtil`. The only possible remaining useful feature is the ability to dump tf-idf vectors (cf. #348) - but pyserini can do this now.

We should just kill this class.

I'm going to leave this issue open for a bit to see if anyone raises objections. Then I'll do it.

### PR DESCRIPTION
**Pull request by lintool**
Closes #1254 

Added `getFieldInfo` and `getFieldInfoDescription` methods to `IndexReaderUtils`, which was some useful functionality I could salvage.

Cleaned up `IndexReaderUtils` as well; removed `dumpDocumentVectors` since we now have that feature in pyserini.

### CODE REVIEW

**Comment by lintool** [comment]
I'm going to wait to see if anyone has objections to this deletion, per #1254 before CR and merge.

**Comment by lintool** [comment]
Regressions are almost finished. Ready for review.

@Chriskamphuis @PepijnBoers Take a look please? Thanks!

File: src/main/java/io/anserini/index/IndexReaderUtils.java [hunk 1]
```diff
@@ -791,4 +672,77 @@ public static String convertLuceneDocidToDocid(IndexReader reader, int docid) {
...
+    if (args.stats) {
+      Terms terms = MultiTerms.getTerms(reader, IndexArgs.CONTENTS);
+
+      System.out.println("Index statistics");
+      System.out.println("----------------");
+      System.out.println("documents:             " + reader.numDocs());
+      System.out.println("documents (non-empty): " + reader.getDocCount(IndexArgs.CONTENTS));
+      System.out.println("unique terms:          " + terms.size());
+      System.out.println("total terms:           " + reader.getSumTotalTermFreq(IndexArgs.CONTENTS));
+    }
```
Comment (PepijnBoers) [inline src/main/java/io/anserini/index/IndexReaderUtils.java:743]: Why not obtain stats from `getIndexStats` and print those? Reduces duplicate code

Comment (lintool) [reply src/main/java/io/anserini/index/IndexReaderUtils.java:743]: oh, good call! 👍

**Pull request merged**